DPTU-58

dbaker006 opened the pull request:
Implement proper handling of ProblemInputView depending on what the current problem is
Remove dependancy on ParentView reference in ProblemInputView.
Add graceful null handling in case the Problem is null when ProblemInputView is called

b-a-merritt committed:
refactor TutoringSessionView update and set modal methods and children

DarrenDevs committed:
DPTU-58 Implement proper handling of ProblemInputView depending on what the current problem is

Removed dependancy on ParentView reference in ProblemInputView.

Added graceful null handling in case the Problem is null when ProblemInputView is called

hasherwi commented:
Be aware that this is probably going to create merge issues: https://dptuapp.atlassian.net/browse/DPTU-56

offqueue commented:
I took a look at your changes today. 
LCSInputView.java: Since you're removing the TutoringSessionView from the constructor for ProblemInputView, you should probably remove the similar constructor from LCSInputView as well. This means that in the submitInput() method around lines 97-98, you'll need to use this.getParent().getParent() in order to use updateWords() and updateStrings(). Right now, you've wrapped these in an "if parentView != null" but it will always be null, so this functionality is basically broken. 
For what it's worth, I had also changed this section of LCSInputView, so at some point we will need to resolve conflicts. Probably better if your code merges first, then I can update mine and reimplement my changes (it's easy enough).

TutoringSessionView updateView() This is minor, but I think there is no longer any reason to remove the ProblemInputView and construct a new one - all the functionality is now in the setModel() method you added. You could just delete lines 284-302. I think?

dbaker006 commented:
Thank you for the suggestions. I've just implemented those changes. Going to re-commit and see what I can do about the merge conflicts.

DarrenDevs committed:
Remove dead code in LCSInputView and replaced it with working logic to
access updateStrings and updateWords

remove code that destroys and reinitializes problemInputView since it is
handled by setModel() function

dbaker006 committed:
Merge branch 'development' into DPTU-58

actions-user committed:
Auto-format code with Spotless

dbaker006 commented:
Resolved merge conflicts

dbaker006 commented:
There is a build error with the current state of the commit. This is part of my pull request but it is actually from @b-a-merritt commit that I accidentally merged with my branch before committing. If you and @hasherwi want to discuss which setModel approach to use

offqueue reviewed (approved):
LGTM
